src/protocol/index.js

```diff
@@ -391,6 +396,21 @@ $root.leancode = (function() {
                 return this.constructor.toObject(this, $protobuf.util.toJSONOptions);
             };
 
+            /**
+             * Gets the default type url for ValueRef
+             * @function getTypeUrl
+             * @memberof leancode.contracts.ValueRef
+             * @static
+             * @param {string} [typeUrlPrefix] your custom typeUrlPrefix(default "type.googleapis.com")
+             * @returns {string} The default type url
+             */
+            ValueRef.getTypeUrl = function getTypeUrl(typeUrlPrefix) {
+                if (typeUrlPrefix === undefined) {
+                    typeUrlPrefix = "type.googleapis.com";
+                }
+                return typeUrlPrefix + "/leancode.contracts.ValueRef";
```

Comment:
This piece of code is repeated in a lot of places. Maybe this could be extracted to some function that takes `typeUrlPrefix` and `typeUrlSuffix` (with `type.googleapis.com` as default `typeUrlPrefix`)? If not then maybe this `if` could be replaced with `??` so it would look like this: `return (typeUrlPrefix ?? "type.googleapis.com") + "/leancode.contracts.ValueRef";` (but I guess this could be less readable)?

Reply:
I didn't know it's auto generated xD